shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java

```diff
@@ -406,6 +416,18 @@ private static PointerCoords parsePointerCoords(Object rawCoords, float density)
         return coords;
     }
 
+    private void validateVirtualDisplayDimensions(int width, int height) {
+        DisplayMetrics metrics = mContext.getResources().getDisplayMetrics();
+        if (height > metrics.heightPixels || width > metrics.widthPixels) {
```

Comment:
I'm missing context here, but wondering why do we have to be that strict? do we never want to allow VDs bigger than the main display? (if that is the case would be useful to add a comment explaining why we want to be that strict)